mne/preprocessing/_regress.py

```diff
@@ -4,28 +4,49 @@
 
 import numpy as np
 
+from ..defaults import _BORDER_DEFAULT
 from ..epochs import BaseEpochs
-from ..io.pick import _picks_to_idx
+from ..io.pick import _picks_to_idx, pick_info
 from ..io.base import BaseRaw
-from ..utils import _check_preload, _validate_type, _check_option, verbose
+from ..utils import (_check_preload, _validate_type, _check_option, verbose,
+                     fill_doc, copy_function_doc_to_method_doc, _check_fname,
+                     _import_h5io_funcs)
+from ..minimum_norm.inverse import _needs_eeg_average_ref_proj
+from ..viz import plot_regression_weights
+from .. import Evoked
 
 
 @verbose
-def regress_artifact(inst, picks=None, picks_artifact='eog', betas=None,
-                     copy=True, verbose=None):
+def regress_artifact(inst, picks=None, *, exclude='bads', picks_artifact='eog',
+                     betas=None, taa=False, proj=True, copy=True,
+                     verbose=None):
     """Remove artifacts using regression based on reference channels.
 
     Parameters
     ----------
     inst : instance of Epochs | Raw
         The instance to process.
     %(picks_good_data)s
+    exclude : list | 'bads'
+        List of channels to exclude from the regression, only used when picking
+        based on types (e.g., exclude="bads" when picks="meg").
+        Specify ``'bads'`` (the default) to exclude all channels marked as bad.
+
+        .. versionadded:: 1.2
     picks_artifact : array-like | str
         Channel picks to use as predictor/explanatory variables capturing
         the artifact of interest (default is "eog").
     betas : ndarray, shape (n_picks, n_picks_ref) | None
         The regression coefficients to use. If None (default), they will be
         estimated from the data.
+    taa : bool
+        Whether to apply TAA correction as described in equation 9 of
```

Comment:
please expand TAA and I would even consider a more explicit parameter name. Any suggestion?